scrunch/datasets.py

```diff
@@ -141,23 +141,30 @@ def _get_dataset(dataset, connection=None, editor=False, project=None):
         else:
             shoji_ds = project.get_dataset(dataset).resource
     else:
-        # search by dataset name
-        try:
-            shoji_ds = root_datasets.by('name')[dataset].entity
-        except KeyError:
-            # search by dataset id
+        # Check if dataset reference is a valid uuid, if so, try to load it
+        # directly
+        if validate_uuid(dataset):
             try:
-                shoji_ds = root_datasets.by('id')[dataset].entity
-            except KeyError:
                 # search by id on any project
-                try:
-                    dataset_url = urljoin(
-                        root.catalogs.datasets, '{}/'.format(dataset))
-                    shoji_ds = root.session.get(dataset_url).payload
-                except Exception:
-                    raise KeyError(
-                        "Dataset (name or id: %s) not found in context."
-                        % dataset)
+                dataset_url = urljoin(
+                    root.catalogs.datasets, '{}/'.format(dataset))
+                shoji_ds = root.session.get(dataset_url).payload
+            except Exception:
```

Comment:
Catch the right exception, will be a `pycrunch.ClientError` with `status` 404, just then `pass` it. Else continue raising it in case something else went wrong. You only want to ignore the exception if it is a 404 Not Found,